File: questions/question_1.3/problem_1.3.rb
def permutation? word1, word2

end

def histo obj
  case
  when obj.class == String
    return obj.split('').inject(Hash.new(0)){ |h,v| h[v] += 1; h }
  else
    return -1
  end
end

p histo 'hello'

File: questions/question_1.3/problem_1.3.js







var histo = function(obj){
  switch (typeof obj){
    case 'string':
      return histo_string(obj)
    default:
      return -1
  };


};

var histo_string = function(word){
  var hash = {}
  for (index = 0; index < word.length; index++){
    if (hash[word[index]]){
      hash[word[index]]++
    } else {
      hash[word[index]] = 1
    }
  }
  return hash
}

console.log(histo('hello'));